### PR TITLE
Feature/use result filter python

### PR DESCRIPTION
**Pull request by huyfififi**
## Description

I hear that it's generally considered good practice to make functions, including main(), return Result<> when there's a possibility of an error occurring during the function's execution.
I'm considering following the pattern for all the function and migrating them one by one. (It would be super helpful if you could work on some).

This pull request updates `filter_python_files` which receives paths and returns only paths of Python files.
- Update the function signature
- Add a test for empty input
- Update main() and tests to comply with the updated function signature

## Checklist

- [x] I have run `cargo fmt`

### CODE REVIEW

File: src/common.rs [hunk 1]
```diff
@@ -16,15 +17,19 @@ pub fn get_all_paths_in_directory(dir: &Path) -> Vec<PathBuf> {
     paths
 }
 
-pub fn filter_python_files(paths: &Vec<PathBuf>) -> Vec<PathBuf> {
+pub fn filter_python_files(paths: &Vec<PathBuf>) -> Result<Vec<PathBuf>, Box<dyn Error>> {
```
Comment (huyfififi) [inline src/common.rs:20]: using dynamic error for now for simplicity. Once we get familiar with Rust, we might want to use a specific type of errors to save memory

File: src/common.rs [hunk 2]
```diff
@@ -68,6 +73,28 @@ mod tests {
     use super::*;
     use std::path::PathBuf;
 
+    fn assert_result_eq<T, E>(result: Result<T, E>, expected: Result<T, E>)
```
Comment (huyfififi) [inline src/common.rs:76]: This helper function compares `Result<>`
